botocore/http_session.py

```diff
@@ -0,0 +1,172 @@
+import os.path
+import logging
+import socket
+from base64 import b64encode
+
+from urllib3 import PoolManager, proxy_from_url, Timeout
+from urllib3.exceptions import NewConnectionError, ProtocolError
+
+from botocore.vendored import six
+from botocore.vendored.six.moves.urllib_parse import unquote
+from botocore.awsrequest import AWSResponse
+from botocore.compat import filter_ssl_warnings, urlparse
+from botocore.exceptions import ConnectionClosedError, EndpointConnectionError
+
+try:
+    from urllib3.contrib import pyopenssl
+    pyopenssl.extract_from_urllib3()
+except ImportError:
+    pass
+
+filter_ssl_warnings()
+logger = logging.getLogger(__name__)
+DEFAULT_TIMEOUT = 60
+MAX_POOL_CONNECTIONS = 10
+# TODO: move the vendored cacert when we drop requests
+DEFAULT_CA_BUNDLE = os.path.join(
```

Comment:
I didn't copy it over as the CA_BUNDLE kinda muddies the diff. I was planning on moving this at the same time we remove the vendored requests directory.

Reply:
>  kinda muddies the diff

Given we won't likely won't remove the vendored requests directory for a while, I'd prefer if this new stuff had minimal dependencies on anything in `vendored/` unless necessary.  That way we can just remove `vendored/` when appropriate.  It will also give customers something to switch to if for whatever reason they're depending on our cacert.pem.

Reply:
Copied the vendored CA Cert and updated this.